main.go

```diff
@@ -77,17 +78,47 @@ func (f *outputVal) Set(v string) error {
 		return nil
 	}
 
-	file, err := os.Create(v)
-	if err != nil {
-		return fmt.Errorf("opening file: %v", v)
+	// We don't want to create the file right now because, if it is in the same module as the target type,
+	// it will interfere with "packages.Load" and it might cause the code generation to fail (go version <= 1.13).
+	// However, we still want to check that we *CAN* create it.
+	_, err := os.Stat(v)
+	if os.IsNotExist(err) {
+		file, err := os.Create(v)
+		if err != nil {
+			return fmt.Errorf("opening file: %v", v)
+		}
+		_ = file.Close()
+		_ = os.Remove(v)
+	} else {
+		// "Touch" the file, which checks the write permission.
+		currentTime := time.Now().Local()
+		err = os.Chtimes(v, currentTime, currentTime)
+		if err != nil {
+			return fmt.Errorf("opening file: %v", v)
+		}
```

Comment:
An easier solution would be to open the file with RDRW|CREATE (but without TRUNC), and then truncate it right before writing. Would you like to explore such a solution?

Reply:
Thank you for the review! I've updated the PR.